backend/api/urls.py

```diff
@@ -4,5 +4,9 @@
 urlpatterns = [
     path("", view=views.index, name="index"),
     path("quizzes/", view=views.quiz_list, name="quizzes-list"),
+    path("quizzes/create/", view=views.quiz_create, name="quiz-create"),
```

Comment:
Hum how about we expose two routes:
```python
# List and create new quiz
path('quizzes')

# GET info about a quiz, Modify a quiz, delete a quiz
path('quizzes/<str:id>')
```

Reply:
What are the advantage(s) of that way of doing ?

__About the advice__: 
Oh, I was not aware of this 🤔 
I'll take that into account from now on. Thanks 😇

Reply:
> What are the advantage(s) of that way of doing ?

It's not about the advantages but best practices. A route is basically a path to a service your app offer. So `https://mydonamain.com/api/quizzes` is about all the services managing `quizzes`. 

- `GET https://mydonamain.com/api/quizzes`: to list all the quizzes
- `POST https://mydonamain.com/api/quizzes` to create a new quiz
- `GET https://mydonamain.com/api/quizzes/<quiz_id>`: to fetch the information of a quiz
- `PATCH https://mydonamain.com/api/quizzes/<quiz_id>`: modify a quiz
- `DELETE https://mydonamain.com/api/quizzes/<quiz_id>`: delete a quiz
- and so on

Reply:
Okayyyy
Juste a question: is it `PATCH` or `PUT`, for modifying a quiz ?
I also read that `PUT` is generally a better method for update than `POST`.
I'm a little confused between all of these

Reply:
Hum yeah those are confusing. I know for sure that `POST` is better for new resource creation. `PUT` is like an upsert. So it's an idempotent method: meaning that if you execute a PUT request more than 1 time, then you will not have any errors. For our case, imagine you made the same CREATE request twice for a quiz... You should get an error, right?

So POST is better than PUT.

PATCH is when you already have the resource, and you want to make a small modification.